### DISCUSSION TITLE
How to add a background image to canvas fabric js 6.0

### DISCUSSION THREAD

**Discussion by sohailahmad07** [fabricjs/fabric.js#10004] [Q&A]
Hi how can i add a background image to a canvas in fabric js 6.0

here my code i don't get any console error but there is not image on the canvas 
```
  fabric.FabricImage.fromURL("http://localhost/storage/img/background.jpg").then(function(img) {
                canvas.backgroundImage = img;
                img.canvas = canvas;

                console.log(canvas.backgroundImage);
            });
```

**Comment by asturur** [comment]
You have to explictly render the canvas with canvas.requestRenderAll() when you do some change.

If that doesn't work you either have the viewport completely out of boundary compared to the image or i do not know.
